[skip ci] Update README.md - Add Featured Models list.

bbeggsTT opened the pull request:
Featured Models list to replace the full model list on the tt-metal repo landing README.

### Ticket
N/A

### Problem description
- Moved original full model list to [models folder location](https://github.com/tenstorrent/tt-metal/blob/main/models/README.md).
- tt-metal README should showcase selected models in great working state

### What's changed
- tt-metal landing README will now display a list of featured models.
- Added a workflow to check if model attributes match between tt-metal/README and tt-metal/models/README every time either file is updated

bbeggsTT committed:
[skip ci] Update README.md - Draft of Featured Models list.

Working draft for Featured Models list to replace the full model list on the tt-metal repo landing README.

bbeggsTT committed:
Update README.md, featured models

Replacing model matrix with featured models.

skhorasganiTT commented on README.md:
```diff
@@ -32,106 +32,44 @@
-| [Mistral 7B](./models/tt_transformers)                        | 32    | [n150](https://tenstorrent.com/hardware/wormhole)        |  101 | 28.3  | 23     | 905.6 | [v0.59.0-rc52](https://github.com/tenstorrent/tt-metal/tree/v0.59.0-rc52)    |  [f028da1](https://github.com/tenstorrent/vllm/tree/f028da11b5b8205272bf18a478de93bd2dd3e29e/tt_metal) |
-| [Mixtral 8x7B (TP=8)](./models/demos/t3000/mixtral8x7b)       | 32    | [QuietBox (Wormhole)](https://tenstorrent.com/hardware/tt-quietbox) | 207       | 16.6  | 33              | 531.2  | [v0.59.0-rc53](https://github.com/tenstorrent/tt-metal/tree/v0.59.0-rc53) |                                                                                                   |
-
-> **Last Update:** June 23, 2025
->
-> **Notes:**
->
-> - ttft = time to first token | t/s/u = tokens/second/user | t/s = tokens/second; where t/s = t/s/u * batch.
-> - TP = Tensor Parallel, DP = Data Parallel; Defines parallelization factors across multiple devices.
-> - The reported LLM performance is for an input sequence length (number of rows filled in the KV cache) of 128 for all models except Mamba (which can accept any sequence length).
-> - The t/s/u reported is the throughput of the first token generated after prefill, i.e. 1 / inter token latency.
-> - Performance numbers were collected using the tt-metal model demos (accessible via the model links). If running with a vLLM inference server, performance may be different.
-> - \* Blackhole software optimization is under active development.  Please join us in shaping the future of open source AI! <br> [\[Discord\]](https://discord.gg/tenstorrent) [\[Developer Hub\]](https://tenstorrent.com/developers)
-> - For more information regarding vLLM installation and environment creation visit the [Tenstorrent vLLM repository](https://github.com/tenstorrent/vllm/blob/dev/tt_metal/README.md).
```
These footnotes are still relevant and should be kept.

bbeggsTT replied:
Footnotes have been included.

skhorasganiTT commented on README.md:
```diff
@@ -32,106 +32,44 @@
+### [Llama 3.1 70B (TP=32)](./models/demos/llama3_subdevices)
+| Batch | Hardware                                                 | ttft (ms) | t/s/u | Target<br>t/s/u | t/s    | TT-Metalium Release                                            | vLLM Tenstorrent Repo Release                                                                                |
+|-------|----------------------------------------------------------|-----------|-------|-----------------|--------|---------------------------------------------------|---------------------------------------------------------------------------------------------------|
+| 32    | [Galaxy (Wormhole)](https://tenstorrent.com/hardware/galaxy) | 105      | 59.1  | 80              | 1891.2  | [v0.59.0-rc52](https://github.com/tenstorrent/tt-metal/tree/v0.59.0-rc52) | [f028da1](https://github.com/tenstorrent/vllm/tree/f028da11b5b8205272bf18a478de93bd2dd3e29e/tt_metal) |
```
Some numbers were updated recently in https://github.com/tenstorrent/tt-metal/commit/a335bb0a2a9ddf4d14388a513a8ec894c021621f, could you apply the same changes here?

bbeggsTT replied:
Release link has been updated.

skhorasganiTT replied:
The entire row needs to be copied (perf numbers and commits)

bbeggsTT replied:
Row updated with new perf numbers/release info.

skhorasganiTT reviewed (commented):
I think reducing the size of the featured model list is a good idea, but one concern I have is that these rows will be duplicated in https://github.com/tenstorrent/tt-metal/blob/main/models/README.md and it will make it harder to perform updates as now both tables will need to be updated. It also increases the risk of the tables diverging. I'm not sure what the thoughts are on having only the featured models on the main page, and having only the "additional models" on the other page (with a note saying where to find the featured models). That's also flawed though since someone might see the second table and miss the featured ones. 
cc @uaydonat

bbeggsTT committed:
[skip ci] Update README.md, 

vLLM repo release info updated, footnotes included.

bbeggsTT committed:
Update README.md

perf numbers/vLLM link updated

bbeggsTT committed:
Update README.md

minor typo

bbeggsTT committed:
Update README.md

Notes moved to top of the model list explaining metrics. Table Column headings capitalized.

bbeggsTT committed:
Update README.md

bbeggsTT commented:
Including @milank94 for any marketing input.

bbeggsTT committed:
Update README.md

bbeggsTT commented on README.md:
```diff
@@ -32,106 +32,55 @@
+
+The models team is focused on developing the following models into a customer ready state; optimizations to achieve performance, accuracy, and compatibility targets are ongoing, follow each model link for more information.
+
+Performance Metrics Notes:
```
I added the LLM performance comment back in.

skhorasganiTT replied:
Thanks, also the part about `(number of rows filled in the KV cache)` and Mamba can be removed

bbeggsTT committed:
[skip ci] Update README.md

bbeggsTT committed:
Update README.md

marty1885 commented:
Thanks - I am aware of this. We have weekly calls regarding content topics. As long as we confirm we confirm we didn't break Model Hub (it pulls from the README table). it is are good.

bbeggsTT commented:
@marty1885 The original model table will be located here (https://github.com/tenstorrent/tt-metal/blob/main/models/README.md) once this PR is pushed.

bbeggsTT committed:
Update README.md

milank94 commented on README.md:
```diff
@@ -32,106 +32,56 @@
+The models team is focused on developing the following models into a customer ready state; optimizations to achieve performance, accuracy, and compatibility targets are ongoing, follow each model link for more information.
+
+Performance Metrics Notes:
+- Time to first token (TTFT) descibes in milliseconds how long it takes for the model to generate the first token.
```
```suggestion
- Time to First Token (TTFT): Measures the time (in milliseconds) it takes to generate the first output token after input is received.
```

bbeggsTT replied:
Changes included.

milank94 commented on README.md:
```diff
@@ -32,106 +32,56 @@
+
+Performance Metrics Notes:
+- Time to first token (TTFT) descibes in milliseconds how long it takes for the model to generate the first token.
+- T/S/U measures tokens per second per user. The T/S/U reported is the throughput of the first token generated after prefill, i.e. 1 / inter token latency.
```
```suggestion
- T/S/U (Tokens per Second per User): Represents the throughput of first-token generation after prefill. It is calculated as 1 / inter-token latency.
```

milank94 commented on README.md:
```diff
@@ -32,106 +32,56 @@
+Performance Metrics Notes:
+- Time to first token (TTFT) descibes in milliseconds how long it takes for the model to generate the first token.
+- T/S/U measures tokens per second per user. The T/S/U reported is the throughput of the first token generated after prefill, i.e. 1 / inter token latency.
+- T/S measures tokens per second, where T/S = T/S/U * batch.
```
```suggestion
- T/S (Tokens per Second): Represents total token throughput, calculated as T/S = T/S/U × batch size.
```

bbeggsTT replied:
Changes included.

milank94 commented on README.md:
```diff
@@ -32,106 +32,56 @@
+- Time to first token (TTFT) descibes in milliseconds how long it takes for the model to generate the first token.
+- T/S/U measures tokens per second per user. The T/S/U reported is the throughput of the first token generated after prefill, i.e. 1 / inter token latency.
+- T/S measures tokens per second, where T/S = T/S/U * batch.
+- Tensor Parallel (TP) and Data Parallel (DP) define parallelization factors across mulitiple devices.
```
```suggestion
- TP (Tensor Parallel) and DP (Data Parallel): Indicate the parallelization factors across multiple devices.
```

milank94 reviewed (changes requested):
Left some suggestions to polish up the points.

BB - Suggestions have been included.

bbeggsTT committed:
Update README.md

Updates from @milank94

bbeggsTT commented on README.md:
```diff
@@ -32,106 +32,56 @@
+- T/S measures tokens per second, where T/S = T/S/U * batch.
+- Tensor Parallel (TP) and Data Parallel (DP) define parallelization factors across mulitiple devices.
+- Reported LLM performance is for an input sequence length of 128 for all models.
+- Performance numbers were collected using the tt-metal model demos (accessible via the model links). If running with a vLLM inference server, performance may be different.
```
Changes included.

bbeggsTT committed:
Update README.md

Re-ordered featured models.

gsarabandoTT commented:
@bbeggsTT For the record: Developers Hub now sources model data from models/README.md. Changes to tt-metal/README.md will no longer impact it.

bbeggsTT committed:
Merge branch 'main' into bbeggsTT-patch-4

bbeggsTT committed:
Update README.md

Conflict Resolved

gsarabandoTT commented:
@bbeggsTT @milank94 I am adding a workflow to check if information between the featured models and the full list still match when either file is edited. This should ensure we always push updates on both places. Let me know what you think about it.

gsarabandoTT committed:
Add models-chart-sync workflow

gsarabandoTT committed:
Merge branch 'main' into bbeggsTT-patch-4

gsarabandoTT committed:
Update README and merge main

gsarabandoTT commented on .github/workflows/models-chart-sync.yaml:
```diff
@@ -0,0 +1,33 @@
+name: Check Featured Models Chart Sync
+
+on:
+  pull_request:
```
Yes! So we can make sure we won't let tables diverge.

bbeggsTT committed:
Merge branch 'main' into bbeggsTT-patch-4

bbeggsTT committed:
Update README.md

Llama 3.1 70B link fixed.

bbeggsTT committed:
Merge branch 'main' into bbeggsTT-patch-4

milank94 commented on .github/scripts/utils/models-chart-sync.py:
Sounds good. I think the links can remain optional. Yeah, I think having minimum format rules is good enough.

milank94 reviewed (approved):
Changes look good to me.

bbeggsTT committed:
Merge branch 'main' into bbeggsTT-patch-4

gsarabandoTT committed:
Remove workflow to split PR

bbeggsTT committed:
Merge branch 'main' into bbeggsTT-patch-4
